src/transformers/cache_utils.py

```diff
@@ -1226,8 +1226,8 @@ def update(
             # `tensor[:, :, index] = tensor`, but the first one is compile-friendly and it does explicitly an in-place
             # operation, that avoids copies and uses less memory.
             try:
-                k_out.index_copy_(2, cache_position, key_states)
-                v_out.index_copy_(2, cache_position, value_states)
+                k_out.index_copy_(2, cache_position, key_states.to(k_out.dtype))
+                v_out.index_copy_(2, cache_position, value_states.to(v_out.dtype))
```

Comment:
You also have to do it in the `NotImplementedError` logic below no ? Maybe do the cast rigth after we computed `k_out` and `v_out`

Reply:
Right, fixed.